modules/hostap/src/supp_main.c

```diff
@@ -214,8 +215,20 @@ static int add_interface(struct supplicant_context *ctx, struct net_if *iface)
 		goto out;
 	}
 
+	wpa_s->conf->okc = 1;
 	wpa_s->conf->filter_ssids = 1;
-	wpa_s->conf->ap_scan = 1;
+
+	if (eth_ctx->eth_if_type == L2_ETH_IF_TYPE_WIFI_STA) {
+		wpa_s->conf->ap_scan = 1;
+	}
+#ifdef CONFIG_AP
+	else if (eth_ctx->eth_if_type == L2_ETH_IF_TYPE_WIFI_UAP) {
+		wpa_s->conf->ap_scan = 2;
+	}
+#endif
```

Comment:
Please move to a separate commit as it's unrelated to this commit log.

Reply:
will update, thanks